File: Solutions/3430/csharp/Program.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace Quera {
    public static class Program {
        public static void Main() {
            var input = GetInputs<string>(1).Single();
            Console.WriteLine(input);

            for (var i = 1; i < input.Length; i++) {
                var targetChar = Convert.ToChar(input.Substring(i, 1));
                var text = new string(targetChar, i); //Repeat char i times.
                text += input.Substring(i); //Adds remain chars
                Console.WriteLine(text);
            }
        }

        private static List<T> GetInputs<T>(int count) {
            var result = new List<T>(count);
            for (var i = 0; i < count; i++) {
                result.Add((T) Convert.ChangeType(Console.ReadLine(), typeof(T)));
            }

            return result;
        }
    }
}

File: Solutions/3430/javascript/program.js
class Solution {
    /**
     * @param {string} str
     */
    solve(str) {
        console.log(str);
        for (let i = 1; i < str.length; i++)
            console.log(`${str[i].repeat(i + 1)}${str.substring(i + 1, str.length)}`)
    }
}

let solution = new Solution();
if (typeof readline === 'function') {
    // This is for Quera judge
    solution.solve(readline());
} else {
    //This is for manual test
    solution.solve("golabi");
    solution.solve("codecup");
}
